tests/test_base_factories.py

```diff
@@ -30,6 +34,47 @@ class MyModel:
 
     class MyFactory(FooDataclassFactory):
         __model__ = MyModel
-        __base_factory_overrides__ = {MyModel: FooDataclassFactory, MyModelWithFoo: FooDataclassFactory}
+        __base_factory_overrides__ = {MyModelWithFoo: FooDataclassFactory}
 
     MyFactory.build()
+
+
+@pytest.mark.parametrize("override_BaseModel", [False, True])
```

Comment:
can you split this into two distinct tests rather than add conditional logic? it will make stuff a bit simpler to comprehend what is actually being tested.